### ISSUE TITLE
Problem with handling failed resumption

### ISSUE THREAD

**Issue by Zash** [dino/dino#1164]
Saw this in server logs after waking up my laptop from suspend:

```xml
# C=Client, S=Server
# Connect, SASL, etc, then
C: <stream:stream>
S: <stream:stream>
S: <stream:features>
    <bind xmlns='urn:ietf:params:xml:ns:xmpp-bind'/>
    <sm xmlns='urn:xmpp:sm:3'/>
  </stream:features>

# Dino tries to resume an expired session
C: <resume previd='8a886656-364b-4f54-91a7-7adfdb4d4f7e' h='51152' xmlns='urn:xmpp:sm:3'>
# Only the 'h' remains of that session.
S: <failed xmlns='urn:xmpp:sm:3' h='7993'>

# Dino tries again?
C: <resume previd='8a886656-364b-4f54-91a7-7adfdb4d4f7e' h='51152' xmlns='urn:xmpp:sm:3'>
S: <failed xmlns='urn:xmpp:sm:3'>

C: <iq type='set' id='800d6709-9fee-46fb-ba90-21f3991281bc'>
    <bind xmlns='urn:ietf:params:xml:ns:xmpp-bind'>
      <resource>dino.xxxx</resource>
    </bind>
  </iq>
S: <iq type='result' id='800d6709-9fee-46fb-ba90-21f3991281bc'>
    <bind xmlns='urn:ietf:params:xml:ns:xmpp-bind'>
      <jid>juliet@im.example.com/dino.xxxx</jid>
    </bind>
  </iq>

# Don't know if relevant
C: <iq type='get' id='2a4bbe36-30ec-45d4-b581-bd726af8c0e3' to='im.example.com'>
S: <iq to='juliet@im.example.com/dino.xxxx' id='2a4bbe36-30ec-45d4-b581-bd726af8c0e3' from='im.example.com' type='result'>

# Stream management is NOT enabled, but Dino sends
C: <r xmlns='urn:xmpp:sm:3'>
S: <stream:error><unsupported-stanza-type xmlns='urn:ietf:params:xml:ns:xmpp-streams'/></stream:error>
S: </stream:stream>
```

Transcript reconstructed from logs.

**Comment by Zash** [comment]
Is `received_features_node` what triggers the retry? Should it perhaps be after the session id is set to `null`?
https://github.com/dino/dino/blob/4e9957deaf4af2258dada078d251524ca96f7675/xmpp-vala/src/module/xep/0198_stream_management.vala#L153-L155

**Comment by fiaxh** [comment]
This was resolved by #1166.
